frame/support/procedural/src/pallet/parse/extra_constants.rs

```diff
@@ -94,6 +119,20 @@ impl ExtraConstantsDef {
 				return Err(syn::Error::new(method.sig.generics.where_clause.span(), msg))
 			}
 
+			// parse constant_name
+			let mut call_var_attrs: Vec<ExtraConstAttr> = helper::take_item_pallet_attrs(method)?;
+
+			// if call_var_attrs.len() > 1 {
+			// 	let msg = "Invalid attribute in pallet::constant_name, only one attribute is expected";
+			// 	return Err(syn::Error::new(call_var_attrs[1].constant_name.span(), msg))
+			// }
+
+			// let constant_name = if call_var_attrs.len() == 1 {
+			// 	call_var_attrs.pop().unwrap().constant_name
+			// } else {
+			// 	method.sig.ident.clone()
+			// };
```

Comment:
You would think this could would absolutely be necessary for the tests to pass.  But they pass without it for some reason.  See for yourself.  

Any thoughts on that?  Can I manually view the Metadata in another way to make sure everything looks good?

Reply:
the test is failing to me:
```
Diff < left / right > :
 PalletMetadata {
     name: "Example",
     storage: Some(
         PalletStorageMetadata {
             prefix: "Example",
             entries: [
                 StorageEntryMetadata {
                     name: "ValueWhereClause",
                     modifier: Optional,
                     ty: Plain(
                         UntrackedSymbol {
                             id: 5,
                             marker: PhantomData,
                         },
                     ),
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "Value",
                     modifier: Optional,
                     ty: Plain(
                         UntrackedSymbol {
                             id: 2,
                             marker: PhantomData,
                         },
                     ),
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "Value2",
                     modifier: Optional,
                     ty: Plain(
                         UntrackedSymbol {
                             id: 5,
                             marker: PhantomData,
                         },
                     ),
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "Map",
                     modifier: Default,
                     ty: Map {
                         hashers: [
                             Blake2_128Concat,
                         ],
                         key: UntrackedSymbol {
                             id: 4,
                             marker: PhantomData,
                         },
                         value: UntrackedSymbol {
                             id: 33,
                             marker: PhantomData,
                         },
                     },
                     default: [
                         4,
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "Map2",
                     modifier: Optional,
                     ty: Map {
                         hashers: [
                             Twox64Concat,
                         ],
                         key: UntrackedSymbol {
                             id: 33,
                             marker: PhantomData,
                         },
                         value: UntrackedSymbol {
                             id: 2,
                             marker: PhantomData,
                         },
                     },
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "DoubleMap",
                     modifier: Optional,
                     ty: Map {
                         hashers: [
                             Blake2_128Concat,
                             Twox64Concat,
                         ],
                         key: UntrackedSymbol {
                             id: 35,
                             marker: PhantomData,
                         },
                         value: UntrackedSymbol {
                             id: 2,
                             marker: PhantomData,
                         },
                     },
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "DoubleMap2",
                     modifier: Optional,
                     ty: Map {
                         hashers: [
                             Twox64Concat,
                             Blake2_128Concat,
                         ],
                         key: UntrackedSymbol {
                             id: 36,
                             marker: PhantomData,
                         },
                         value: UntrackedSymbol {
                             id: 5,
                             marker: PhantomData,
                         },
                     },
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "NMap",
                     modifier: Optional,
                     ty: Map {
                         hashers: [
                             Blake2_128Concat,
                         ],
                         key: UntrackedSymbol {
                             id: 4,
                             marker: PhantomData,
                         },
                         value: UntrackedSymbol {
                             id: 2,
                             marker: PhantomData,
                         },
                     },
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "NMap2",
                     modifier: Optional,
                     ty: Map {
                         hashers: [
                             Twox64Concat,
                             Blake2_128Concat,
                         ],
                         key: UntrackedSymbol {
                             id: 36,
                             marker: PhantomData,
                         },
                         value: UntrackedSymbol {
                             id: 5,
                             marker: PhantomData,
                         },
                     },
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "RenamedCountedMap",
                     modifier: Optional,
                     ty: Map {
                         hashers: [
                             Twox64Concat,
                         ],
                         key: UntrackedSymbol {
                             id: 4,
                             marker: PhantomData,
                         },
                         value: UntrackedSymbol {
                             id: 2,
                             marker: PhantomData,
                         },
                     },
                     default: [
                         0,
                     ],
                     docs: [],
                 },
                 StorageEntryMetadata {
                     name: "CounterForRenamedCountedMap",
                     modifier: Default,
                     ty: Plain(
                         UntrackedSymbol {
                             id: 2,
                             marker: PhantomData,
                         },
                     ),
                     default: [
                         0,
                         0,
                         0,
                         0,
                     ],
                     docs: [
                         "Counter for the related counted storage map",
                     ],
                 },
             ],
         },
     ),
     calls: Some(
         PalletCallMetadata {
             ty: UntrackedSymbol {
                 id: 37,
                 marker: PhantomData,
             },
         },
     ),
     event: Some(
         PalletEventMetadata {
             ty: UntrackedSymbol {
                 id: 39,
                 marker: PhantomData,
             },
         },
     ),
     constants: [
         PalletConstantMetadata {
             name: "MyGetParam",
             ty: UntrackedSymbol {
                 id: 2,
                 marker: PhantomData,
             },
             value: [
                 10,
                 0,
                 0,
                 0,
             ],
             docs: [
                 " Some comment",
                 " Some comment",
             ],
         },
         PalletConstantMetadata {
             name: "MyGetParam2",
             ty: UntrackedSymbol {
                 id: 2,
                 marker: PhantomData,
             },
             value: [
                 11,
                 0,
                 0,
                 0,
             ],
             docs: [
                 " Some comment",
                 " Some comment",
             ],
         },
         PalletConstantMetadata {
             name: "MyGetParam3",
             ty: UntrackedSymbol {
                 id: 5,
                 marker: PhantomData,
             },
             value: [
                 12,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
             ],
             docs: [],
         },
         PalletConstantMetadata {
             name: "some_extra",
             ty: UntrackedSymbol {
                 id: 5,
                 marker: PhantomData,
             },
             value: [
                 100,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
             ],
             docs: [
                 " Some doc",
                 " Some doc",
             ],
         },
         PalletConstantMetadata {
             name: "some_extra_extra",
             ty: UntrackedSymbol {
                 id: 5,
                 marker: PhantomData,
             },
             value: [
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
             ],
             docs: [
                 " Some doc",
             ],
         },
         PalletConstantMetadata {
<            name: "some_extra_rename",
>            name: "SomeExtraRename",
             ty: UntrackedSymbol {
                 id: 5,
                 marker: PhantomData,
             },
             value: [
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
                 0,
             ],
             docs: [
                 " Some doc",
             ],
         },
     ],
     error: Some(
         PalletErrorMetadata {
             ty: UntrackedSymbol {
                 id: 40,
                 marker: PhantomData,
             },
         },
     ),
     index: 1,
 }

', frame/support/test/tests/pallet.rs:1525:5
```

Reply:
I must be running the wrong test command:
`cargo test pallet.rs`

What did you run to get this result?

Reply:
cargo test -p frame-support-test --test pallet

Reply:
Thanks!  That's the issue then.